### ISSUE TITLE
Add ability to inject properties into alert messages

### ISSUE THREAD

**Issue by whoiskevinrich** [bytenik/Seq.App.Slack#26]
Allow the insertion of seq properties into messages parsed by alerts

**Comment by rocklan** [comment]
The problem is that alerts are an aggregate of multiple messages, so how would you choose which properties to include?

**Comment by whoiskevinrich** [comment]
The alert is fired from a query. By default, the alert trigger is something like `count > 5`.

![image](https://user-images.githubusercontent.com/2496620/45589019-4df03c00-b8d3-11e8-8716-666aa1a8bb3f.png)

![image](https://user-images.githubusercontent.com/2496620/45589016-3dd85c80-b8d3-11e8-9281-b1b536a8135d.png)

However, alerting supports customizing the query to fire based on multiple select statements: 
![image](https://user-images.githubusercontent.com/2496620/45589057-04542100-b8d4-11e8-855b-715cde33470c.png)

To which the Alerts system has access to:
![image](https://user-images.githubusercontent.com/2496620/45589059-1209a680-b8d4-11e8-9c05-98c73f8a402c.png)

Now the example above is contrived, and I would probably never do anything useful with an Applications Property, but sometimes I'm tracking to make sure a signal property is staying within bounds, and I'd like to know what the Min and Max value of a timeslice is, base my alarm on that, and do so without creating separate alarms.

Basically, I'm not looking for the properties of the signals, but rather the properties on the dashboard query's "Select" statement

**Comment by nblumhardt** [comment]
These are provided to the app in a property called `Results`, which is more like a rowset than easily-readable property values, unfortunately.

It's used in the email app's default template here: https://github.com/datalust/seq-apps/blob/master/src/Seq.App.EmailPlus/Resources/DefaultBodyTemplate.html#L84

This currently only shows group headings rather than column headings :-/ but does include the data:

![image](https://user-images.githubusercontent.com/342712/45590447-2f924b00-b97c-11e8-84a2-ef31777ef7a1.png)

Not sure what kinds of display options Slack provides, but we might be able to format this up nicely as a table with a bit of tinkering...

**Comment by dazinator** [comment]
Also really would like this.

I wonder if one possibility would be to "Group By" for example "Tenant Name"
Then allow the GroupBy argument to be accessible as a property in the template?

For example if we are grouping by TenantName, then it would be good if we could somehow use the group's value in the template e.g "Customer A"

![Image](https://github.com/user-attachments/assets/b011001a-6dbd-439f-b090-ae1c21e586bb)

**Comment by whoiskevinrich** [comment]
I'm closing the issue; it's 7 years old and I'm not concerned about it anymore :D

**Comment by nblumhardt** [comment]
@dazinator this might also fall out of #68, if we were to do that - group names are in the `Results` object and expressions can be used to pull them out. It's a little trickier because one alert might be triggered by multiple groups, but still seems to be roughly aligned with that direction...
